src/formatDate.js

```diff
@@ -8,7 +8,61 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitElement = fromFormat[3];
+  const joinElement = toFormat[3];
+  const newDate = date.split(splitElement);
+
+  let formatAgan = [];
+
+  let year = formatAgan[0];
+  let month = formatAgan[1];
+  let day = formatAgan[2];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YY' || fromFormat[i] === 'YYYY') {
+      year = newDate[i];
+    }
+
+    if (fromFormat[i] === 'YYYY') {
```

Comment:
The check for 'YYYY' is redundant here since it's already included in the previous if condition that checks for both 'YY' and 'YYYY'. You can remove this block to simplify the code.